python/ray/data/_internal/execution/operators/zip_operator.py

```diff
@@ -0,0 +1,238 @@
+import itertools
+from typing import Callable, List, Optional, Tuple
+
+import ray
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
+from ray.data._internal.remote_fn import cached_remote_fn
+from ray.data._internal.split import _split_at_indices
+from ray.data._internal.stats import StatsDict
+from ray.data._internal.execution.interfaces import (
+    RefBundle,
+    PhysicalOperator,
+)
+from ray.data.block import (
+    Block,
+    BlockAccessor,
+    BlockExecStats,
+    BlockMetadata,
+    BlockPartition,
+)
+
+
+class ZipOperator(PhysicalOperator):
+    """An operator that zips its inputs together."""
```

Comment:
Comment that this implementation is bulk for now.

Reply:
@ericl - added.